tscreen.go

```diff
@@ -939,17 +939,15 @@ func (t *tScreen) enableMouse(f MouseFlags) {
 	if len(t.mouse) != 0 {
 		// start by disabling all tracking.
 		t.TPuts("\x1b[?1000l\x1b[?1002l\x1b[?1003l\x1b[?1006l")
-		if f&MouseButtonEvents != 0 {
-			t.TPuts("\x1b[?1000h")
-		}
-		if f&MouseDragEvents != 0 {
-			t.TPuts("\x1b[?1002h")
-		}
 		if f&MouseMotionEvents != 0 {
-			t.TPuts("\x1b[?1003h")
+			// Some terminals (putty, hterm, etc.) may ignore 1003,
+			// in which case use 1002 instead.
```

Comment:
Ok I think I understand what happened here.

The 1006h should *not* have been set on leaving the application.  That's the only bug here.

The rest of this change is inappropriate.  I can refactor this to be correct if need be, or if you want to resubmit so that we only send the 1006h if we sent *other* settings (basically if f &(MouseMotionEvents|....) != 0 { ... })

Reply:
Terminal will notify all type of mouse events in 1003 mode.
If user only wants motion events, then button and drag events should be ignored.
Or tcell should provide that feature.

MouseMotionEvents is defined on [screen.go](https://github.com/gdamore/tcell/blob/0d961d2608bb074c4995ed9e278e33f8263636d4/screen.go#L264) as follows:

```go
MouseMotionEvents = MouseFlags(4) // All mouse events (includes click and drag events)
```

So it seems that using MouseMotionEvents with the intention of getting only motion events is not in the current spec.

...

Anyway, the purpose of this MR is not to set 1006 when the application is closed, so I would like to reset it to commit [39161d6](https://github.com/gdamore/tcell/pull/512/commits/39161d638ec0367bf2a898d89d9ce7f59acdb9f0).

Reply:
Completed reset to [39161d6](https://github.com/gdamore/tcell/pull/512/commits/39161d638ec0367bf2a898d89d9ce7f59acdb9f0)